src/Config/ConfigOption.php

```diff
@@ -0,0 +1,290 @@
+    {
+        return $this->scopes;
+    }
+
+    /**
+     * @return ConfigSection The most immediate section of the option (it may be a child of a parent section)
+     */
+    public function getSection(): ConfigSection
+    {
+        return $this->section;
+    }
+
+    /**
+     * @return ConfigSection[] The full path of sections to the option
+     */
+    public function getSections(): array
+    {
+        $sections = [];
+        $section = $this->section;
+        while ($section !== null) {
+            $sections[] = $section;
+            $section = $section->getParent();
+        }
+        return array_reverse($sections);
+    }
+
+    /**
+     * @return string The internal name of the option
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return string The label of the option
+     */
+    public function getLabel(): string
+    {
+        return $this->label;
+    }
+
+    /**
+     * @return string The internal name of the option (context.name)
+     */
+    public function getFullName(): string
+    {
+        return $this->getContext() . '.' . $this->getName();
+    }
+
+    /**
+     * @return ConfigType The type of the option
+     */
+    public function getType(): ConfigType
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return array The options for the type of the option
+     */
+    public function getTypeOptions(): array
+    {
+        return $this->type_options;
+    }
+
+    /**
+     * @return string The context of the option
+     */
+    public function getContext(): string
+    {
+        return $this->context;
+    }
+
+    /**
+     * Get the value of the configuration option for a specific scope.
+     * @param ConfigScope $scope The scope to get the value for
+     * @param mixed|null $default The default value to return if the option value is not found
+     * @param array $scope_params The parameters to use when looking up the option value
+     * @phpstan-param array{context?: string, entities_id?: int, users_id?: int} $scope_params
+     * @return mixed
+     */
+    public function getValue(ConfigScope $scope, mixed $default = null, array $scope_params = []): mixed
+    {
+        if (!in_array($scope, $this->scopes, true)) {
+            return $default;
+        }
+        $fn_format_loaded_value = function ($value) {
+            if (isset($this->type_options['multiple']) && $this->type_options['multiple']) {
+                return importArrayFromDB($value);
+            }
+            return $value;
+        };
+        switch ($scope) {
+            case ConfigScope::GLOBAL:
+                return $fn_format_loaded_value(\Config::getConfigurationValue($scope_params['context'] ?? 'core', $this->getName())) ?? $default;
+            case ConfigScope::ENTITY:
+                $entity = new \Entity();
+                if (!$entity->getFromDB($scope_params['entities_id'] ?? \Session::getLoginUserID())) {
```

Comment:
I think we should force passing the `entities_id` param here. Falling back to session value may hide misusage of the method.
Same remarks applies also for `ConfigScope::USER` and for the `setValue()` and the `getComputedValue()` methods.